transport/src/main/java/io/netty/channel/BackoffStrategyFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+/**
+ * Factory that creates a new {@link BackoffStrategy} every time.
+ */
+public interface BackoffStrategyFactory<T extends BackoffStrategy> {
+
+    /**
+     * Creates a new {@link BackoffStrategy}.
+     */
+    T newBackoffStrategy();
```

Comment:
Does having a generic factory provide any value in this case? Can we just return `BackoffStrategy`?